apps/rush-lib/src/data/ChangeManagement.ts

```diff
@@ -80,3 +87,87 @@ export interface IChangeInfo {
    */
   type?: string;
 }
+
+/**
+ * Writes the change file to disk in sync mode.
+ * @internal
+ *
+ * @param rushConfiguration - Rush configuration
+ * @param changeFile - change file data
+ */
+export function _writeChangeFileSync(
+  rushConfiguration: RushConfiguration,
+  changeFile: IChangeFile
+): void {
+  const filePath: string = _generateChangeFilePath(rushConfiguration, changeFile.packageName);
+  fsx.ensureFileSync(filePath);
+  fsx.writeFileSync(filePath, JSON.stringify(changeFile, undefined, 2));
+}
+
+/**
+ * Generates change file path based on the provided information.
+ * @internal
+ *
+ * @param rushConfiguration - Rush configuration
+ * @param packageName - package name
+ */
+export function _generateChangeFilePath(
+  rushConfiguration: RushConfiguration,
+  packageName: string
+): string {
+  let branch: string = undefined;
+  try {
+    branch = gitInfo().branch;
+  } catch (error) {
+    console.log('Could not automatically detect git branch name, using timestamp instead.');
+  }
+
+  // example filename: yourbranchname_2017-05-01-20-20.json
+  const filename: string = (branch ?
+    this._escapeFilename(`${branch}_${this._getTimestamp()}.json`) :
+    `${this._getTimestamp()}.json`);
+  const filePath: string = path.join(this.rushConfiguration.commonFolder,
+    RushConstants.changeFilesFolderName,
+    ...packageName.split('/'),
+    filename);
+  return filePath;
+}
+
+/**
+* Gets the current time, formatted as YYYY-MM-DD-HH-MM
+* Optionally will include seconds
+*/
+function _getTimestamp(useSeconds: boolean = false): string {
+  // Create a date string with the current time
+
+  // dateString === "2016-10-19T22:47:49.606Z"
+  const dateString: string = new Date().toJSON();
+
+  // Parse out 2 capture groups, the date and the time
+  const dateParseRegex: RegExp = /([0-9]{4}-[0-9]{2}-[0-9]{2}).*([0-9]{2}:[0-9]{2}:[0-9]{2})/;
+
+  // matches[1] === "2016-10-19"
+  // matches[2] === "22:47:49"
+  const matches: RegExpMatchArray = dateString.match(dateParseRegex);
+
+  // formattedDate === "2016-10-19"
```

Comment:
>formattedDate [](start = 5, length = 13)

We should be consistent in our style... For example, should we be exporting public methods with _ ? I would prefer to see this wrapped in a static class, but what we really need to do is have consistency. We should adopt a style guide similar to SPFx-Core.